lib/supavisor/monitoring/telem.ex

```diff
@@ -3,69 +3,102 @@ defmodule Supavisor.Monitoring.Telem do
 
   require Logger
 
-  alias Supavisor, as: S
+  @metrics_disabled Application.compile_env(:supavisor, :metrics_disabled, false)
 
-  @spec network_usage(:client | :db, S.sock(), S.id(), map()) :: {:ok | :error, map()}
+  defmacro if_metrics_enabled(do: block) do
```

Comment:
I think that instead of creating such macro, that applies code only if the metrics are enabled, I would create macro:

```elixir
defmacro telemetry_execute(event_name, measurements, metadata) do
  if not @metrics_disabled do
    quote do
      :telemetry.execute(unquote(event_name), unquote(measurements), unquote(metadata))
    end
  end
end
```

That would IMHO be easier to read.

Reply:
yeah, agree